src/PostgREST/Main.hs

```diff
@@ -71,6 +67,10 @@ main = do
         fail "Cannot run in this PostgreSQL version, PostgREST needs at least 9.2.0"
     ) supportedOrError
 
+  Right authenticator <- H.session pool $ do
```

Comment:
As suspicious as it looks this might be the right way to do it. The hasql library does not actually parse out individual settings that you could ask for (it sends `StringSettings` right to the db and actually converts `ParamSettings` to a string prior to sending to the db, see [1](https://github.com/nikita-volkov/hasql-postgres/blob/6a87068af052d1de6c6a22297621f3df7933423b/library/Hasql/Postgres/Connector.hs#L73)).

The pattern `Right foo <- bar` is a partial pattern match which can fail at runtime. Except in our case if it fails then we are in a seriously weird situation and it should fail. Maybe the only improvement here is to handle `Right authenticator` and `Left e` separately. In the left case print a potentially more explanatory message to the user and die.